objects/src/accounts/account_id.rs

```diff
@@ -127,6 +136,133 @@ impl AccountId {
 
     /// Finds and returns a seed suitable for creating an account ID for the specified account type
     /// using the provided initial seed as a starting point.
+    #[cfg(feature = "std")]
+    pub fn get_account_seed(
+        init_seed: [u8; 32],
+        account_type: AccountType,
+        on_chain: bool,
+        code_root: Digest,
+        storage_root: Digest,
+    ) -> Result<Word, AccountError> {
+        let thread_count = thread::available_parallelism().map_or(1, |v| v.get());
+
+        let (send, recv) = mpsc::channel();
+        let stop = OnceLock::new();
+
+        for count in 0..thread_count {
+            let send = send.clone();
+            let stop = stop.clone();
+            let mut init_seed = init_seed;
+            init_seed[0] += count as u8;
+            spawn(move || {
+                Self::get_account_seed_inner(
+                    send,
+                    stop,
+                    init_seed,
+                    account_type,
+                    on_chain,
+                    code_root,
+                    storage_root,
+                )
+            });
+        }
+
+        #[allow(unused_variables)]
+        let (digest, seed) = recv.recv().unwrap();
+        let _ = stop.set(true);
+
+        #[cfg(feature = "log")]
+        {
+            log::info!(
+                "Found account seed [pow={}, digest={}, seed={}]",
+                digest_pow(digest),
+                digest_hex(digest),
+                word_hex(seed),
+            );
+        }
+        Ok(seed)
+    }
+
+    #[cfg(feature = "std")]
+    pub fn get_account_seed_inner(
+        send: Sender<(Digest, Word)>,
+        stop: OnceLock<bool>,
+        init_seed: [u8; 32],
+        account_type: AccountType,
+        on_chain: bool,
+        code_root: Digest,
+        storage_root: Digest,
+    ) {
+        let init_seed: Vec<[u8; 8]> =
+            init_seed.chunks(8).map(|chunk| chunk.try_into().unwrap()).collect();
+        let mut current_seed: Word = [
+            Felt::from(init_seed[0]),
+            Felt::from(init_seed[1]),
+            Felt::from(init_seed[2]),
+            Felt::from(init_seed[3]),
+        ];
+        let mut current_digest = Self::compute_digest(current_seed, code_root, storage_root);
+
+        #[cfg(feature = "log")]
+        {
+            log::info!(
+                "Generating new account seed [pow={}, digest={}, seed={}]",
+                digest_pow(current_digest),
+                digest_hex(current_digest),
+                word_hex(current_seed),
+            );
+        }
+
+        let mut count = 0;
+        #[cfg(feature = "log")]
+        let mut best: (u32, Digest, Word) =
+            (digest_pow(current_digest), current_digest, current_seed);
```

Comment:
I think you can move `best` to the curly brackets above such that we have a single feature flag.

Reply:
ref: https://github.com/0xPolygonMiden/miden-base/pull/223#discussion_r1315335534